### PR TITLE
Added a Warning When VS2019 is not Installed with Link to Download it

### PR DESCRIPTION
**Pull request by AMZN-nggieber**
o3de_o5jzPDauOJ

### CODE REVIEW

**Comment by AMZN-ScottR** [comment]
A bit of a nitpick here but could we widen the dialog a bit?  Looks cramped for the amount of text on display.

**Review by AMZN-ScottR — APPROVED** [review]

**Comment by Engineered4Fun** [comment]
I solved this by restarting my pc.. still can't get engine to run though.

**Review by AMZN-Phil — APPROVED** [review]

**Review by AMZN-alexpete — CHANGES_REQUESTED** [review]
I think we should PALify this, and use some kind of generic ProjectUtils function called FindSupportedCompiler() that will return an AZ::Outcome that contains success or an error message you can display in a MessageBox.

The Windows implementation can call IsVS2019Installed() for now.

The Mac/Linux implementations can always return success or we can use an AZ_TRAIT define to disable the check on those platforms till we have the code to detect XCode and clang etc.

**Comment by AMZN-ScottR** [comment]
> I think we should PALify this, and use some kind of generic ProjectUtils function called FindSupportedCompiler() that will return an AZ::Outcome that contains success or an error message you can display in a MessageBox.
> 
> The Windows implementation can call IsVS2019Installed() for now.
> 
> The Mac/Linux implementations can always return success or we can use an AZ_TRAIT define to disable the check on those platforms till we have the code to detect XCode and clang etc.

Have to agree with the PAL-ification here

**Review by AMZN-ScottR — APPROVED** [review]
Thanks for doing the PAL-ification!

File: Code/Tools/ProjectManager/Platform/Windows/ProjectUtils_windows.cpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                        "-latest",
+                        "-requires",
+                        "Microsoft.VisualStudio.Component.VC.Tools.x86.x64",
+                        "-property",
+                        "isComplete"
+                    });
+
+                if (vsWhereProcess.waitForStarted())
+                {
+                    while (vsWhereProcess.waitForReadyRead()) {}
```
Comment (AMZN-alexpete) [inline Code/Tools/ProjectManager/Platform/Windows/ProjectUtils_windows.cpp:45]: A sleep() or usleep() might be appropriate in this while loop

File: Code/Tools/ProjectManager/Platform/Windows/ProjectUtils_windows.cpp [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                    QString vsWhereOutput(vsWhereProcess.readAllStandardOutput());
+                    if (vsWhereOutput.startsWith("1"))
```
Comment (AMZN-alexpete) [inline Code/Tools/ProjectManager/Platform/Windows/ProjectUtils_windows.cpp:47]: hmmm should we not be using https://doc.qt.io/qt-5/qprocess.html#exitCode here?

File: Code/Tools/ProjectManager/Source/ProjectUtils.cpp [hunk 3]
```diff
@@ -369,46 +371,27 @@ namespace O3DE::ProjectManager
...
-                }
+                QMessageBox vsWarningMessage(parent);
+                vsWarningMessage.setIcon(QMessageBox::Warning);
+                vsWarningMessage.setWindowTitle(QObject::tr("Create Project"));
+                // Makes link clickable
+                vsWarningMessage.setTextFormat(Qt::RichText);
+                vsWarningMessage.setText(findCompilerResult.GetError());
+                vsWarningMessage.setStandardButtons(QMessageBox::Close);
+
+                QSpacerItem* horizontalSpacer = new QSpacerItem(600, 0, QSizePolicy::Minimum, QSizePolicy::Expanding);
```
Comment (AMZN-alexpete) [inline Code/Tools/ProjectManager/Source/ProjectUtils.cpp:388]: NIT: IMHO this is a bit too wide,  maybe 400-450?

**Review by AMZN-alexpete — APPROVED** [review]
Just a couple minor suggestions

File: Code/Tools/ProjectManager/Platform/Windows/ProjectUtils_windows.cpp (hunk 1 above)
Comment (AMZN-nggieber) [reply Code/Tools/ProjectManager/Platform/Windows/ProjectUtils_windows.cpp]: Replaced with waitForFinished

**Pull request merged**